### PR TITLE
Implement better password validation #147

### PR DESCRIPTION
**Pull request by Te-k**
Here is a proposal implementation for #147 using zxcvbn

### CODE REVIEW

File: OpenOversight/app/templates/auth/register.html [hunk 1]
```diff
@@ -9,8 +9,82 @@
...
+    var password2 = document.getElementById('password2');
+    var meter = document.getElementById('password-strength-meter');
+    var meter2 = document.getElementById('password-confirmation-meter');
+    var text = document.getElementById('password-strength-text');
+
+    function check_confirmation() {
+        var val1 = password.value;
+        var val2 = password2.value;
+        console.log(val1);
+        console.log(val2);
```
Comment (redshiftzero) [inline OpenOversight/app/templates/auth/register.html]: Can remove these two `console.log` lines

Comment (Te-k) [inline OpenOversight/app/templates/auth/register.html]: Oops, yes I will :)

**Comment by Te-k** [comment]
Nope, I will be happy to do both this week :)

**Comment by redshiftzero** [comment]
Awesome! Feel free to ping me for feedback or a re-review 💪

**Comment by Te-k** [comment]
Here it is : 
* Code migrated into a separated js file
* Same client side check on /change-password
* Form requirement of 8 password minimum character (I also changed login to 6 min, 8 is really long, is it ok?)
* No more console.log :stuck_out_tongue:

**Comment by redshiftzero** [comment]
Nice!! Changes look great, just tested them out. Also :+1: to changing login to 6 minimum, I'm cool with that. 

One last thing, it still looks like the password strength is not being used to disallow registration of accounts with weak passwords - for example, I was able to register an account with the password `password` - legitimately scored as "Very Bad" by the code. But since it was 8 characters or greater none of the form validators threw an error.

**Comment by Te-k** [comment]
Yes, the client side is only showing an indication, the only forced constraint is the 8 char minimum server side. I thought it was your idea with zxcvbn, did you want to block the form client side ?

It is possible, I just need to work a bit on it to be sure that the reason is clear for the user.

**Comment by Te-k** [comment]
I just improved the js to disable the submit button if the password is not strong enough. If someone can check that I don't fall into some weird js implementation on their browser it would be cool.

**Comment by freddymartinez9** [comment]
It looks good to me! I'm approving for merge.

**Comment by redshiftzero** [comment]
Tested the changes in Firefox, Safari, and Chrome. Looks great!! Thanks for this @Te-k! Gonna merge in...

**Comment by redshiftzero** [comment]
(Merged in these commits manually to resolve merge conflicts)

**Pull request closed without merging**